plugin/storage/badger/spanstore/writer.go

```diff
@@ -67,9 +68,22 @@ func (w *SpanWriter) WriteSpan(_ context.Context, span *model.Span) error {
 		return err
 	}
 
+	kind, _ := span.GetSpanKind()
+
 	entriesToStore = append(entriesToStore, trace)
 	entriesToStore = append(entriesToStore, w.createBadgerEntry(createIndexKey(serviceNameIndexKey, []byte(span.Process.ServiceName), startTime, span.TraceID), nil, expireTime))
-	entriesToStore = append(entriesToStore, w.createBadgerEntry(createIndexKey(operationNameIndexKey, []byte(span.Process.ServiceName+span.OperationName), startTime, span.TraceID), nil, expireTime))
+	entriesToStore = append(
+		entriesToStore,
+		w.createBadgerEntry(
+			createOperationWithKindIndexKey(
+				[]byte(span.Process.ServiceName+span.OperationName),
```

Comment:
please pass these separately. Is there any other way to call createOperationWithKindIndexKey?